### PR TITLE
add documentation for new endpoint projectItems.updateInternalHourlyCost

### PR DESCRIPTION
**Pull request by joaoinacio**
⚠️ DO NOT MERGE - should be under "profit on projects" feature flag.

Document new endpoint which allows specifying a project participant's internal rate for a project, or all projects.

e2e tests: https://github.com/teamleadercrm/api-tests/pull/609
Implementation: https://github.com/teamleadercrm/core/pull/9365

### CODE REVIEW

**Review by ketlane — DISMISSED** [review]

File: src/07-projects/projects.apib [hunk 1]
```diff
@@ -226,3 +226,18 @@ Update a participant's role for a project.
...
+
+### projects.updateInternalHourlyCost [POST /projects.updateInternalHourlyCost]
+
+Updates the internal hourly cost for a participant for one project, or all projects.
+This does not update the global internal rate of the employee. See `users.updateInternalHourlyCost`.
+
++ Request (application/json;charset=utf-8)
+
+    + Attributes (object)
+        + id: `082e6289-30c5-45ad-bcd0-190b02d21e81` (string, optional) - If there is no specific project id, update for all projects
```
Comment (adognameddog) [inline src/07-projects/projects.apib]: This should be required, we update the internal rate for a participant for one project. That would be consistent with the other endpoints.

**Review by adognameddog — DISMISSED** [review]

File: src/07-projects/projects.apib (hunk 1 above)
Comment (ketlane) [reply src/07-projects/projects.apib]: But how would you update the internal rate for all projects (for a deactivated/deleted user)?

File: src/07-projects/projects.apib [hunk 2]
```diff
@@ -230,13 +230,13 @@ Update a participant's role for a project.
...
 
-Updates the internal hourly cost for a participant for one project, or all projects.
+Updates the internal hourly cost for a participant for one project.
 This does not update the global internal rate of the employee. See `users.updateInternalHourlyCost`.
 
 + Request (application/json;charset=utf-8)
 
     + Attributes (object)
-        + id: `082e6289-30c5-45ad-bcd0-190b02d21e81` (string, optional) - If there is no specific project id, update for all projects
+        + id: `082e6289-30c5-45ad-bcd0-190b02d21e81` (string, required)
```
Comment (adognameddog) [inline src/07-projects/projects.apib]: I don't know if you have to specify "required". Leaving out "optional" is enough.

**Review by adognameddog — DISMISSED** [review]

**Review by ketlane — DISMISSED** [review]

**Review by ArnaudWeyts — DISMISSED** [review]

File: src/07-projects/project-items.apib [hunk 3]
```diff
@@ -134,3 +134,17 @@ Update a project item's internal cost.
         + internal_cost_per_unit (Money, required, nullable)
 
 + Response 204 (application/json;charset=utf-8)
+
+### projectItems.updateInternalHourlyCost [POST /projectItems.updateInternalHourlyCost]
```
Comment (ketlane) [inline src/07-projects/project-items.apib:138]: Q: Shouldn't this be internal hourly rate, instead of internal hourly cost?

Comment (ketlane) [reply src/07-projects/project-items.apib:138]: See my comment in the implementation PR: https://github.com/teamleadercrm/core/pull/9365#discussion_r276610350

Comment (joaoinacio) [reply src/07-projects/project-items.apib:138]: Agree in theory, lets discuss on slack ;)

Comment (ketlane) [reply src/07-projects/project-items.apib:138]: We didn't find a better name. Closing this remark :D

**Review by ketlane — DISMISSED** [review]

**Review by ArnaudWeyts — DISMISSED** [review]

**Review by adognameddog — DISMISSED** [review]

**Review by jenssegers — DISMISSED** [review]

**Review by jenssegers — APPROVED** [review]

File: src/07-projects/project-items.apib [hunk 4]
```diff
@@ -133,4 +133,18 @@ Update a project item's internal cost.
...
+
+Updates the internal hourly cost for a participant, for a specific project or all projects. Only account administrators have access to this endpoint.
+This does not update the global internal rate of the employee. See `users.update`.
+
++ Request (application/json;charset=utf-8)
+
+    + Attributes (object)
+        + employee_id: `b5094b3f-bb7a-0391-b01b-e709773f3509` (string, required)
```
Comment (stivni) [inline src/07-projects/project-items.apib:146]: PP : user_id

File: src/07-projects/projects.apib (hunk 2 above)
Comment (stivni) [reply src/07-projects/projects.apib]: PP : I like the required to be present, it's more explicit

**Review by ArnaudWeyts — APPROVED** [review]

**Review by adognameddog — APPROVED** [review]

**Pull request merged**